src/theory/sets/theory_sets_private.cpp

```diff
@@ -501,45 +288,32 @@ void TheorySetsPrivate::fullEffortCheck(){
   do{
     Trace("sets") << "...iterate full effort check..." << std::endl;
     Assert( d_equalityEngine.consistent() );
-    d_sentLemma = false;
-    d_addedFact = false;
     d_full_check_incomplete = false;
-    d_set_eqc.clear();
-    d_set_eqc_list.clear();
-    d_eqc_emptyset.clear();
-    d_eqc_univset.clear();
-    d_eqc_singleton.clear();
-    d_congruent.clear();
-    d_nvar_sets.clear();
-    d_var_set.clear();
     d_most_common_type.clear();
```

Comment:
Wouldn't it be nice to have a reset function for TheorySetsPrivate similar to SolverState and others ?

Reply:
Will do.